rust/azure_iot_operations_services/src/azure_device_registry.rs

```diff
@@ -5,20 +5,179 @@
 
+/// Represents a Device in the Azure Device Registry service.
+pub struct Device {
+    /// The 'name' Field.
+    pub name: String,
+    /// The 'specification' Field.
+    pub specification: DeviceSpecification,
+    /// The 'status' Field.
+    pub status: Option<DeviceStatus>,
+}
+
+#[derive(Debug, Clone)]
+pub struct DeviceSpecification {
+    /// The 'attributes' Field.
+    pub attributes: HashMap<String, String>, // if None, we can represent as empty hashmap
+    /// The 'discoveredDeviceRef' Field.
+    pub discovered_device_ref: Option<String>,
+    /// The 'enabled' Field.
+    pub enabled: Option<bool>,
+    /// The 'endpoints' Field.
+    pub inbound_endpoints: HashMap<String, InboundEndpoint>, // if None, we can represent as empty hashmap
+    /// The 'externalDeviceId' Field.
+    pub external_device_id: Option<String>,
+    /// The 'lastTransitionTime' Field.
+    pub last_transition_time: Option<String>, // DateTime?
+    /// The 'manufacturer' Field.
+    pub manufacturer: Option<String>,
+    /// The 'model' Field.
+    pub model: Option<String>,
+    /// The 'operatingSystem' Field.
+    pub operating_system: Option<String>,
+    /// The 'operatingSystemVersion' Field.
+    pub operating_system_version: Option<String>,
+    /// The 'uuid' Field.
+    pub uuid: Option<String>,
+    /// The 'version' Field.
+    pub version: Option<u64>,
+}
+
+#[derive(Debug, Clone)]
+pub struct InboundEndpoint {
+    /// The 'additionalConfiguration' Field.
+    pub additional_configuration: Option<String>,
+    /// The 'address' Field.
+    pub address: String,
+    /// The 'authentication' Field.
+    pub authentication: Authentication,
+    /// The 'trustSettings' Field.
+    pub trust_settings: Option<TrustSettings>,
+    /// The 'type' Field.
+    pub r#type: String,
+    /// The 'version' Field.
+    pub version: Option<String>,
+}
+
+#[derive(Debug, Clone)]
+pub struct TrustSettings {
+    /// The 'issuerList' Field.
+    pub issuer_list: Option<String>,
+    /// The 'trustList' Field.
+    pub trust_list: Option<String>,
+    /// The 'trustMode' Field.
+    pub trust_mode: String,
+}
+
+#[derive(Debug, Clone)] // default Anonymous
+pub enum Authentication {
+    Anonymous,
+    Certificate {
+        /// The 'certificateSecretName' Field.
+        certificate_secret_name: String,
+    },
+    UsernamePassword {
+        /// The 'passwordSecretName' Field.
+        password_secret_name: String,
+        /// The 'usernameSecretName' Field.
+        username_secret_name: String,
+    },
+}
+// ~~~~~~~~~~~~~~~~~~~Device Endpoint Status DTDL Equivalent Structs~~~~
+#[derive(Clone, Debug, Default)]
+/// Represents the status of a Device in the ADR Service.
+pub struct DeviceStatus {
+    /// The 'config' Field.
+    pub config: Option<DeviceStatusConfig>,
+    /// The 'endpoints' Field.
+    pub endpoints: HashMap<String, Option<ConfigError>>,
+}
+
+#[derive(Clone, Debug, Default)]
+pub struct DeviceStatusConfig {
```

Comment:
I think this can eventually be shared with the asset structs (from asset.yaml), they have the same fields, so maybe it can go in common

Reply:
ooh yep, I'll change that now so Oliva can use the same one - good catch!